storey/steps/assertion.py

```diff
@@ -0,0 +1,149 @@
+from dataclasses import dataclass
+from typing import Callable, List, Any, Collection
+
+from storey.dtypes import _termination_obj
+from storey.flow import Flow
+
+
+@dataclass
+class _Operator:
+    str: str
+    fn: Callable[[Any, Any], bool]
+
+    def __call__(self, x, y):
+        return self.fn(x, y)
+
+    def __str__(self):
+        return self.str
+
+
+_EQUALS = _Operator("==", lambda x, y: x == y)
+_NOT_EQUALS = _Operator("!=", lambda x, y: x != y)
+_ABOVE = _Operator(">", lambda x, y: x > y)
+_BELOW = _Operator("<", lambda x, y: x < y)
+_ABOVE_OR_EQUAL = _Operator(">=", lambda x, y: x >= y)
+_BELOW_OR_EQUAL = _Operator("<=", lambda x, y: x <= y)
+
+_ANY = _Operator("any of", lambda x, y: any([i for i in x if i in y]))
```

Comment:
If checking for membership, should it be `any([i in y for i in x])`? Otherwise a member of x could be a value that evaluates to `False` (e.g. `False`, `0`, `''`).

Reply:
It's also a bit confusing that ANY and ALL mean "any of the members in x are members of y" and "all members of x are members of y". It seems like these are set operations, so normally these would be refered to in terms of intersection and union and subsets.

Reply:
hmm how about python's `any` and `all` functions?

Reply:
Those are boolean operations rather than set comparison operations.

Reply:
Unfortunately intersect, union and disjoint are not very concise in this case, hopefully my changes made it better now